backend/apps/github/common.py

```diff
@@ -91,8 +92,61 @@ def sync_repository(gh_repository, organization=None, user=None):
                     issue.labels.add(Label.update_data(gh_issue_label))
                 except UnknownObjectException:
                     logger.info("Couldn't get GitHub issue label %s", issue.url)
+    # GitHub Pull Requests Part!
     else:
         logger.info("Skipping issues sync for %s", repository.name)
+    if not repository.is_archived and repository.project:
+        kwargs = {
+            "direction": "asc",
+            "sort": "created",
+            "state": "open",
+        }
+
+        open_gh_pull_requests = list(gh_repository.get_pulls(**kwargs))
+        open_pr_numbers = {gh_pull_request.number for gh_pull_request in open_gh_pull_requests}
+
+        closed_prs = PullRequest.objects.filter(repository=repository, state="open").exclude(
+            number__in=open_pr_numbers
+        )
+
+        for pr in closed_prs:
+            pr.state = "closed"
+            try:
+                gh_closed_pr = gh_repository.get_pull(pr.number)
```

Comment:
This is too expensive -- it's a call per each PR. 
I think we should have 2 loops over open and closed PRs separately. This way we'll be able to get those closed_at/merge_at attributes in a more efficient way.